### DISCUSSION TITLE
webauthn.dll usage and python-fido2

### DISCUSSION THREAD

**Discussion by tobiaszheller** [Yubico/libfido2#635] [Q&A] [answered]
Hello,
I am working now on integration with windows hello as fido device and one thing is bothering me. 

libfido2.c library supports windows hello via webauthn.dll integration, right?
Why does https://github.com/Yubico/python-fido2 use dll directly instread of libfido2.c API?

**Comment by dainnilsson** [comment, marked as answer]
Interacting with an Authenticator on Windows without requiring Administrator access has to go through the webauthn.h API, which is why both libfido2 and python-fido2 support optionally doing so. Can you elaborate on why you think python-fido2 should take a detour via libfido2 to achieve this?

**Reply by tobiaszheller** [reply]
@dainnilsson Thanks for fast response and sorry for not being clear in first place.

As far as I understand yubico/libfido2 uses webauthn.dll on windows platform. (https://github.com/Yubico/libfido2/blob/4329ad8de4b37d07113f8940c7150c0176d16ede/src/dev.c#L261-L278)

So calling without admin privileges  `fido_dev_info_manifest` should return both yubi devices and windows hello device right?
Unfortunately in my case yubi device is only returned when I run it with admin privileges, win hello is returned always.

While trying to debug it I have noticed that `yubico/python-fido2` does not use `fido_dev_info_manifest` from c library on windows.  Instead it uses `webauthn.dll` directly https://github.com/Yubico/python-fido2/blob/cd9e6cbd595118e93d38a9d2535e36576df9abb9/examples/credential.py#L57. 
Do you know why is it so?

**Reply by dainnilsson** [reply]
I'm sorry, but I still don't understand why you would have the expectation that python-fido2 should use libfido2 instead of webauthn.h directly. To me it's akin to asking why wget doesn't use curl, or why Firefox doesn't use Chrome. They are different projects. Two separate implementations of FIDO functionality, with some of the same goals, but also different ones.

**Reply by tobiaszheller** [reply]
Ah sorry. I was sure that https://github.com/Yubico/python-fido2 under the hood uses https://github.com/Yubico/libfido2 but now I can see that it is direct implementation. Sorry for confusion, we can close that discussion.

**Reply by dainnilsson** [reply]
Ah! then your question makes a lot more sense! Sorry for the confusion! 😄
